integrations/aws/aws/session_manager.py

```diff
@@ -0,0 +1,161 @@
+                            # Replace the Temp account details with the current account details
+                            self._aws_accessible_accounts[0] = account
+                            continue
+                        await self._assume_role_and_update_credentials(
+                            sts_client, account
+                        )
+            except organizations_client.exceptions.AccessDeniedException:
+                logger.error(
+                    "Caller is not a member of an AWS organization. Assuming role in the current account."
+                )
+        logger.info(f"Found {len(self._aws_credentials)} AWS accounts")
+
+    async def _assume_role_and_update_credentials(
+        self, sts_client: STSClient, account: dict[str, Any]
+    ) -> None:
+        try:
+            account_role = await sts_client.assume_role(
+                RoleArn=f'arn:aws:iam::{account["Id"]}:role/{self._get_account_read_role_name()}',
+                RoleSessionName="AssumeRoleSession",
+            )
+            raw_credentials = account_role["Credentials"]
+            credentials = AwsCredentials(
+                account_id=account["Id"],
+                access_key_id=raw_credentials["AccessKeyId"],
+                secret_access_key=raw_credentials["SecretAccessKey"],
+                session_token=raw_credentials["SessionToken"],
+            )
+            await credentials.update_enabled_regions()
+            self._aws_credentials.append(credentials)
+            self._aws_accessible_accounts.append(account)
+        except sts_client.exceptions.ClientError as e:
+            if e.response["Error"]["Code"] == "AccessDenied":
+                logger.info(f"Cannot assume role in account {account['Id']}. Skipping.")
+                pass  # Skip the account if assume_role fails due to permission issues or non-existent role
+            else:
+                raise
+
+    async def find_account_id_by_session(self, session: aioboto3.Session) -> str:
+        session_credentials = await session.get_credentials()  # type: ignore
+        frozen_credentials = await session_credentials.get_frozen_credentials()
+        for cred in self._aws_credentials:
+            if cred.access_key_id == frozen_credentials.access_key:
+                return cred.account_id
+
+        if len(self._aws_credentials) == 1:
+            return self._aws_credentials[0].account_id
+
+        raise ValueError(f"Cannot find credentials linked with this session {session}")
+
+    def find_credentials_by_account_id(self, account_id: str) -> AwsCredentials:
+        for cred in self._aws_credentials:
+            if cred.account_id == account_id:
+                return cred
+
+        if len(self._aws_credentials) == 1:
+            return self._aws_credentials[0]
+
+        raise ValueError(
+            f"Cannot find credentials linked with this account id {account_id}"
+        )
```

Comment:
where do you handle all of those errors?
maybe we should create custom error classes?

Reply:
I do not handle this error, how would a custom error class help in this kind of scenario?